packages/transformers/js/src/JSTransformer.js

```diff
@@ -706,51 +706,64 @@ export default (new Transformer({
         asset.symbols.set(exported, local, convertLoc(loc));
       }
 
-      let deps = new Map(
-        asset
-          .getDependencies()
-          .map(dep => [dep.meta.placeholder ?? dep.specifier, dep]),
-      );
-      for (let dep of deps.values()) {
-        dep.symbols.ensure();
+      let deps = new Map();
+      for (let dep of asset.getDependencies()) {
+        if (!deps.has(dep.meta.placeholder ?? dep.specifier)) {
+          deps.set(dep.meta.placeholder ?? dep.specifier, []);
+        }
+        deps.get(dep.meta.placeholder ?? dep.specifier)?.push(dep);
+      }
+      for (let depArr of deps.values()) {
+        for (let dep of depArr) {
+          dep.symbols.ensure();
+        }
       }
 
       for (let {
         source,
         local,
         imported,
         loc,
+        kind,
       } of hoist_result.imported_symbols) {
-        let dep = deps.get(source);
-        if (!dep) continue;
-        dep.symbols.set(imported, local, convertLoc(loc));
+        for (let dep of nullthrows(deps.get(source))) {
+          if (
+            (dep.meta.kind === 'Require' ||
+              dep.meta.kind === 'Import' ||
+              dep.meta.kind === 'DynamicImport') &&
+            dep.meta.kind !== kind
+          ) {
+            continue;
+          }
+          dep.symbols.set(imported, local, convertLoc(loc));
+        }
       }
-
       for (let {source, local, imported, loc} of hoist_result.re_exports) {
-        let dep = deps.get(source);
-        if (!dep) continue;
-
-        if (local === '*' && imported === '*') {
-          dep.symbols.set('*', '*', convertLoc(loc), true);
-        } else {
-          let reExportName =
-            dep.symbols.get(imported)?.local ??
-            `$${asset.id}$re_export$${local}`;
-          asset.symbols.set(local, reExportName);
-          dep.symbols.set(imported, reExportName, convertLoc(loc), true);
+        for (let dep of nullthrows(deps.get(source))) {
+          if (local === '*' && imported === '*') {
+            dep.symbols.set('*', '*', convertLoc(loc), true);
+          } else {
```

Comment:
I think this shouldn't be set for all dependencies with that specified for only for the `Import` one?

Reply:
So we shouldn't set the symbols if `kind` is `Import`?